django/thunderstore/api/cyberstorm/views/community.py

```diff
@@ -0,0 +1,214 @@
+                    & Q(
+                        Case(
+                            When(
+                                require_package_listing_approval=True,
+                                then=Case(
+                                    When(
+                                        Q(
+                                            package_listings__review_status=PackageListingReviewStatus.approved
+                                        ),
+                                        then=True,
+                                    ),
+                                    default=False,
+                                    output_field=BooleanField(),
+                                ),
+                            ),
+                            default=Case(
+                                When(
+                                    Q(
+                                        package_listings__review_status=PackageListingReviewStatus.unreviewed
+                                    )
+                                    | Q(
+                                        package_listings__review_status=PackageListingReviewStatus.approved
+                                    ),
+                                    then=True,
+                                ),
+                                default=False,
+                                output_field=BooleanField(),
+                            ),
+                        ),
+                    ),
+                ),
+                distinct=True,
+            ),
+            downloads=Sum(
+                Case(
+                    When(package_listings__package__is_deprecated=True, then=0),
+                    When(
+                        package_listings__review_status=PackageListingReviewStatus.rejected,
+                        then=0,
+                    ),
+                    default="package_listings__package__versions__downloads",
+                )
+            ),
+        )
+
+    def order_queryset(
+        self, ordering: str, queryset: QuerySet[Community]
+    ) -> QuerySet[Community]:
+        order_arg = {
+            "name": "name",
+        }.get(ordering, "name")
+
+        return queryset.order_by(Lower(order_arg))
+
+    def paginate(
+        self, params: OrderedDict, queryset: QuerySet[Community], page_size: int
+    ) -> Page:
+        paginator = CachedPaginator(
+            queryset,
+            page_size,
+            cache_key="api.communities.paginator",
+            cache_vary=self.get_full_cache_vary(params),
+            cache_bust_condition=CacheBustCondition.any_package_updated,
+        )
+
+        # PageNotAnInteger error won't be raised here since deserializer
+        # has already checked that the page parameter is an integer.
+        try:
+            page = paginator.page(params["page"])
+        except EmptyPage:
+            raise ParseError("Page index error: no results on requested page")
+
+        return page
```

Comment:
No need to use cached paginator due to what was mentioned regarding caching. Rely on the DRF provided ones if they're sufficient, but in general you probably don't need to deal with pagination directly yourself if you rely on the DRF-provided features around pagination.